docs: add security explanation page (ISD-5931) (#129)
canonical/content-cache-operator: docs/explanation/index.md

---
myst:
  html_meta:

>     "description lang=en": "Explanation of the Content Cache Operators design, architecture, and how they work under the hood."

Reviewer: ```suggestion
    "description lang=en": "Explanation of the Content Cache Operators design, architecture, and security."
```
nit, think it would be good to include something about security in the metadata
